### PR TITLE
Allow to mark spam assets as false positive

### PR DESCRIPTION
**Pull request by yabirgb**
Backend part for https://github.com/orgs/rotki/projects/11?pane=issue&itemId=45931334

### CODE REVIEW

File: rotkehlchen/tests/api/test_assets.py [hunk 1]
```diff
@@ -1055,3 +1065,55 @@ def test_only_ignored_assets(rotkehlchen_api_server):
     )
     result = assert_proper_response_with_result(response)
     assert {entry['identifier'] for entry in result['entries']} == set(ignored_assets)
+
+
+def test_false_positive(rotkehlchen_api_server: APIServer, globaldb: GlobalDBHandler) -> None:
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_assets.py:1070]: docstring explaining what this test tests for

File: rotkehlchen/tests/api/test_assets.py [hunk 2]
```diff
@@ -1055,3 +1065,55 @@ def test_only_ignored_assets(rotkehlchen_api_server):
...
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
+    db = rotki.data.db
+
+    # setup as if dai was a spam assets
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_assets.py:1074]:
```suggestion
    # setup as if DAI was in the spam assets
```

File: rotkehlchen/tasks/assets.py [hunk 3]
```diff
@@ -142,7 +143,17 @@ def augmented_spam_detection(user_db: DBHandler) -> None:
             f'SELECT identifier FROM assets WHERE identifier IN ({", ".join(["?"] * len(assets))}) AND type=?',  # noqa: E501
             [*assets, AssetType.EVM_TOKEN.serialize_for_db()],
         )
+
+        # query assets marked as false positive
```
[the remark this answers is not in the document]
Comment (yabirgb) [reply rotkehlchen/tasks/assets.py:147]: That happens only in the get_or_create_evm_token so the assets is not there in the database yet. I believe it doesn't make sense

File: rotkehlchen/api/rest.py [hunk 4]
```diff
@@ -4635,3 +4640,44 @@ def list_accounting_rules_conflicts(self, filter_query: DBFilterQuery) -> Respon
...
+        with self.rotkehlchen.data.db.user_write() as write_cursor:  # remove it from the ignored assets  # noqa: E501
+            self.rotkehlchen.data.db.remove_from_ignored_assets(
+                write_cursor=write_cursor,
+                asset=token,
+            )
+
+        return api_response(OK_RESULT, status_code=HTTPStatus.OK)
+
+    def remove_from_spam_assets_false_positives(self, token: EvmToken) -> Response:
+        """Delete a spam assets from the list of whitelisted assets"""
```
Comment (LefterisJP) [inline rotkehlchen/api/rest.py:4675]:
```suggestion
        """Delete a spam asset from the list of whitelisted assets"""
```

File: rotkehlchen/globaldb/cache.py [hunk 5]
```diff
@@ -55,23 +55,40 @@ def globaldb_set_general_cache_values(
...
 
 
+def globaldb_delete_general_cache_values(
+        write_cursor: DBCursor,
+        key_parts: Iterable[str | GeneralCacheType],
+        values: tuple[str] | None = None,
+) -> None:
+    """
+    Delete an entry from the general_cache. If any value is provided the rows fow which
+    the key matches the and the value is in the list of values are deleted.
```
Comment (LefterisJP) [inline rotkehlchen/globaldb/cache.py:73]: After the rows fow (which should be for) it makes no sense.

```suggestion
    wtf? :D
```

**Review by LefterisJP — APPROVED** [review]
LGTM. But have 2 nitpicks that should be fixed.

**Pull request merged**